src/controllers/controllerpresetfilehandler.cpp

```diff
@@ -13,56 +13,53 @@
 #include "controllers/hid/hidcontrollerpresetfilehandler.h"
 
 // static
-ControllerPresetPointer ControllerPresetFileHandler::loadPreset(const QString& pathOrFilename,
-                                                                const QStringList& presetPaths) {
-    qDebug() << "Searching for controller preset" << pathOrFilename
-             << "in paths:" << presetPaths.join(",");
-    QString scriptPath = ControllerManager::getAbsolutePath(pathOrFilename,
-                                                            presetPaths);
-
-    if (scriptPath.isEmpty()) {
-        qDebug() << "Could not find" << pathOrFilename
-                 << "in any preset path.";
+ControllerPresetPointer ControllerPresetFileHandler::loadPreset(
+        const QFileInfo& presetFile, const QDir& systemPresetsPath) {
+    VERIFY_OR_DEBUG_ASSERT(presetFile.exists() && presetFile.isReadable()) {
+        qDebug() << "Preset" << presetFile.absoluteFilePath()
+                 << "does not exist or is unreadable.";
         return ControllerPresetPointer();
     }
 
-    QFileInfo scriptPathInfo(scriptPath);
-    if (!scriptPathInfo.exists() || !scriptPathInfo.isReadable()) {
-        qDebug() << "Preset" << scriptPath << "does not exist or is unreadable.";
-        return ControllerPresetPointer();
-    }
-
-    ControllerPresetFileHandler* pHandler = NULL;
-    if (scriptPath.endsWith(MIDI_PRESET_EXTENSION, Qt::CaseInsensitive)) {
+    ControllerPresetFileHandler* pHandler = nullptr;
```

Comment:
I don't think a pointer needs to be explicitly initialized to nullptr?

Reply:
I think it does. It's not always assigned a value below.

Reply:
okay